### PR TITLE
Read bearer token from the cookie token header

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
Allows reading the bearer token from a `Cookie` request header. Our UI already saves the token in a cookie named `token` and it is sent to the server as a part of `Cookie` header.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/auth/oidc.go [hunk 1]
```diff
@@ -232,6 +233,37 @@ func (e queryParamBearerTokenExtractor) Extract(r *http.Request) (string, error)
...
+func NewCookieHeaderBearerTokenExtractor(header string) TokenExtractor {
+	return cookieHeaderBearerTokenExtractor{name: header}
+}
+
+type cookieHeaderBearerTokenExtractor struct {
+	name string
+}
+
+func (e cookieHeaderBearerTokenExtractor) Extract(r *http.Request) (string, error) {
+	header := r.Header.Get("Cookie")
```
Comment (alvaroaleman) [inline api/pkg/handler/auth/oidc.go]: The `*http.Request` has a `Cookie` func, please use that instead of manually parsing the header

Comment (floreks) [reply api/pkg/handler/auth/oidc.go]: Nice! I'll remember that.

**Comment by floreks** [comment]
Done. PTAL

**Review by alvaroaleman — APPROVED** [review]
/lgtm
/approve

**Pull request merged**